src/game/Object.cpp

```diff
@@ -518,7 +518,18 @@ void Object::BuildValuesUpdate(uint8 updatetype, ByteBuffer* data, UpdateMask* u
                         }
                     }
                     else
-                        *data << uint32(0);                 // disable quest object
+                    {
+                        // flag (no interact) quest object when player not eligible for quest
+                        if (gameObject->GetGoType() == GAMEOBJECT_TYPE_QUESTGIVER &&
+                            !gameObject->ActivateToQuest(target) && !gameObject->GetGOInfo()->GetGossipMenuId())
+                        {
+                            updateMask->SetBit(GAMEOBJECT_DYN_FLAGS);
+                            *data << uint16(GO_DYNFLAG_LO_NO_INTERACT);
+                            *data << uint16(0);
+                        }
+                        else
+                            *data << uint32(0);              // disable quest object
```

Comment:
Very small thing but should move this comment to line 520 as it applies to the whole else statement.